### ISSUE TITLE
Create an Animated Autostart and QuickStart

### ISSUE THREAD

**Issue by wlbragg** [c172p-team/c172p#1334]
After watching several aircraft go through their animated autostart's, basically a live "tutorial", would it be something we should offer in the c172p? Have a "QuickStart", that is our current autostart. And also have an animated "autiostart" that actually slowly steps through the physical actions of setting up the aircraft interior panel switches and knobs and starting then cranking the starter until running?
I know it is redundant not only to the current autostart but also the tutorial. But it is a pretty cool thing to watch and acts as a quick lesson to learn to start the aircraft and also autostart it at the same time. Only because the aircraft ships with FG, I feel it should have ever everything there is to offer and as examples to others as to the capabilities in development.
Is this something we might want to do?

**Comment by gilbertohasnofb** [comment]
That sounds like a nice idea. Do you know some good example of other aircraft with this feature?

**Comment by wkitty42** [comment]
On 8/26/20 2:47 PM, wlbragg wrote:
> After watching several aircraft go through their animated autostart's, basically 
> a live "tutorial", would it be something we should offer in the c172p?


you saw that F-16 startup video, eh? :)


> Have a "QuickStart", that is our current autostart. And also have an animated 
> "autiostart" that actually slowly steps through the physical actions of setting 
> up the aircraft interior panel switches and knobs and starting then cranking the 
> starter until running?


i like the idea but prefer it a little slower and more zoomed in than in the 
F-16 video so one can actually see and focus on which switches are being flipped 
and buttons pushed... i had to back that f-16 video up numerous times to be able 
to figure out what was being flipped and pushed and i still miss some things...

an autostart that shows everything would help one to learn the motions and a 
quickstart might be no longer needed... just run the autostart... unless one is 
really that impatient??

**Comment by wkitty42** [comment]
On 8/26/20 3:36 PM, Gilberto Agostinho wrote:
> That sounds like a nice idea. Do you know some good example of other aircraft 
> with this feature?

F-16 video linked on the forums earlier today... i forget which sub-forum, though :/

**Comment by wlbragg** [comment]
https://forum.flightgear.org/viewtopic.php?f=4&t=2003&start=345#p372767

I've seen at least one other that was nice but don't remember where.

**Comment by gilbertohasnofb** [comment]
That's really neat. In our case, it's just a matter of changing properties sequentially with a delay between them instead of all of them at once, right? Basically something along the lines of:

* close doors
* remove control lock
* fuel tanks to BOTH
* throttle to 1/4 open
* mixture to full lean
* prime 3x times
* master on
* move ignition switch to start
* advance mixture in to full rich
* set altimeter pressure 

With like a second or two between each action.

**Comment by gilbertohasnofb** [comment]
How did the F16 devs implement theirs? We use a Nasal function for the autostart, and all properties are set at once.

**Comment by wkitty42** [comment]
at least that much delay, yes... we should be able to use timing to make it about like a human would do if they were doing it manually...

i don't know how the F-16 was done... i'm sure they would spill the beans if asked, though :wink:

**Comment by wlbragg** [comment]
>How did the F16 devs implement theirs?

They didn't, it is someone else that did it and they asked if the F16 devs want the "nasal" code they did it in.

**Comment by gilbertohasnofb** [comment]
> They didn't, it is someone else that did it

That's quite a literal answer :rofl: Ok, let me rephrase it: how did the person who implement it do it? Are the delays between actions done through Nasal?

**Comment by dany93** [comment]
@gilbertohasnofb wrote

> Do you know some good example of other aircraft with this feature?

The Boeing 707 Lake of Constance Hangar :: M.Kraus has it.
It is a bit long to observe.

@wkitty42 
a quickstart might be no longer needed... just run the autostart... unless one is
really that impatient??

I don't agree.
Generally, too long and useless after a few observations. If someone uses the autostart, it means he gives priority to flying. No matter the full realism. Let's let people have the choice..
And I imagine, when I do several startups for testing the aircraft....

**Comment by gilbertohasnofb** [comment]
@dany93 You have a good point, Dany. The only counterarguments I can think of are: 1) it will still take just a couple of seconds for the animation to take place, since the 172P is super simple to start (unlike teh F-16) and 2) new users who ignore tutorials (or don't know about them) will learn from watching it. I myself don't have strong feelings about either implementing it or not.

**Comment by wlbragg** [comment]
>That's quite a literal answer

Sorry, I just meant the F16 team didn't do it yet. It was an offer from someone else. who created it.
In the post they stated...
>If you like, I can make the code changes to f16.nas (and config.xml) available.

So I think the answer is yes, it was done in nasal.

Well this isn't something we have to decide on right now. I could live without it as well as have it. As it is the only aircraft that ships with FG I feel it might be of benefit to a few new users. If we do decide it's better to have than not, I would definitely want to keep the quick autostart. It's not something I want to watch more than one or twice in order to learn about the aircraft.

**Comment by gilbertohasnofb** [comment]
> Sorry, I just meant the F16 team didn't do it yet. 

No need to apologise, I just found it funny! :smile: 

> If we do decide it's better to have than not, I would definitely want to keep the quick autostart.

That's one thing I really dislike. I think I already wrote it a lot of times, but I find that open-source software tend to have too many options in order to satisfy too many people, and the GUI as well as other forms of interaction with the users tend to get really messy. I think it's either one or the other, as having two functions doing the same with the sole difference of activating an animation or not seems like an overkill (and code hoarding :rofl:). The idea of doing an animation for x amount of times then reverting back to our current autostart seems very cryptic to me, users will be baffled why something stopped behaving as they expect, and every update (or loss of the folder `.fgfs`) will 'solve' the 'problem'. 

The thing is: the animation will take at most 5-10 seconds, since it's just a matter of setting throttle mixture, priming if necessary, moving a couple of switches and turning the key; are we willing to live with that or not?

**Comment by wlbragg** [comment]
I'm inclined to think if I had only one choice, I would opt in for the animated version. It's quick enough, and it still saves me the motions of having to go through the steps. But I think we will piss off some regular users? What about a choice of one or the other. Or allow "sequenced start" to be shut off and converted to autostart (no animations) with a hidden property switch or even not so hidden choice in the GUI. But then were back to overkill (and code hoarding 🤣):).

**Comment by wlbragg** [comment]
I guess I kind of just repeated my original post about a choice, just a different way to choose.

**Comment by gilbertohasnofb** [comment]
I think that an option in the aircraft menu called 'animate autostart' (or something along those lines) would definitely be less worse than having two function, one autostart function and another one, say, sequential start function. Personally, I still would prefer to have a single one. For the users, I bet that most people would not be put off by the extra 5-10 seconds added to starting, and most would appreciate. It's mostly devs trying things out who would be a bit less happy with it.

@legoboyvdlp @onox @wkitty42 @Juanvvc any opinions on this?

**Comment by legoboyvdlp** [comment]
I think we have enough options, really. I agree with the single option, as long as it doesn't grow to ~30 seconds like the f16's (old) one. (I haven't tried the new one yet).

**Comment by dany93** [comment]
Why not a separate item in the "Help" Menu? It's not an 'aircraft' option, rather a help, like the Aircraft Checklists...

**Comment by onox** [comment]
> The Boeing 707 Lake of Constance Hangar, M.Kraus, has it.
It is a bit long to observe.

The 707 has in addition a quick autostart (high pressure ground start via the overhead panel) which takes just a few seconds. But this one only exists because the regular autostart takes a few minutes :sleeping: 

For the c172p if the animated autostart only takes 5 - 10 seconds, then a single option is sufficient.

**Comment by wkitty42** [comment]
how long does it really take to manually flip the switches, check the guages, and do the thing in RL?

**Comment by wlbragg** [comment]
After letting time go by and revisiting this, I think a/the startup tutorial is enough instruction if someone needs it. The in sim autostart needs to be fast and is not really meant to be informational. That is what tutorials are for.
